arch/arc/core/reset.S

```diff
@@ -86,6 +86,24 @@ SECTION_SUBSEC_FUNC(TEXT,_reset_and__start,__start)
 	kflag r0
 #endif
 
+#if defined (CONFIG_CPU_HS4X) && defined (CONFIG_ARC_HAS_DSP)
+	/* case of HS4XD cpu */
+	lr	 r0, [_ARC_V2_LPB_BUILD]
+	breq r0, 0, skip_lpb_setup
+	mov  r0, 1	/* disable LPB for HS4XD */
+	sr   r0, [_ARC_V2_LPB_CTRL]
+skip_lpb_setup:
+#endif
+
+#ifdef CONFIG_ARC_HAS_DSP
+	lr	 r0, [_ARC_V2_DSP_BUILD]
+	bmsk r0, r0, 7
+	breq r0, 0, skip_dsp_setup
+	mov	 r0, 0 /* DSP_CTRL_DISABLED_ALL */
+	sr	 r0, [_ARC_V2_DSP_CTRL]
+skip_dsp_setup:
+#endif /* CONFIG_ARC_HAS_DSP */
```

Comment:
We should move it to macro to https://github.com/zephyrproject-rtos/zephyr/blob/main/arch/arc/core/dsp/

And this seems to interfere with existing DSP functionality.